### PR TITLE
[FrameworkBundle] avoid service id conflicts with Swiftmailer

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #31906
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -1916,7 +1916,7 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
         }
 
         $loader->load('mailer.xml');
-        $container->getDefinition('mailer.transport')->setArgument(0, $config['dsn']);
+        $container->getDefinition('mailer.default_transport')->setArgument(0, $config['dsn']);
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: `mailer.transport` would conflict with https://github.com/symfony/monolog-bundle/blob/master/DependencyInjection/Compiler/AddSwiftMailerTransportPass.php#L40 if `$mailerId` is `mailer`

**Review by fabpot — APPROVED** [review]

**Comment by xabbuh** [comment]
alternatively, we can update the compiler pass in MonologBundle and check the transport class there

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer.xml [hunk 2]
```diff
@@ -6,23 +6,23 @@
...
         <service id="mailer.mailer" class="Symfony\Component\Mailer\Mailer">
-            <argument type="service" id="mailer.transport" />
+            <argument type="service" id="mailer.default_transport" />
             <argument type="service" id="messenger.default_bus" on-invalid="ignore" />
         </service>
         <service id="mailer" alias="mailer.mailer" />
         <service id="Symfony\Component\Mailer\MailerInterface" alias="mailer.mailer" />
 
-        <service id="mailer.transport" class="Symfony\Component\Mailer\Transport\TransportInterface">
+        <service id="mailer.default_transport" class="Symfony\Component\Mailer\Transport\TransportInterface">
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer.xml]: changing the id here might cause trouble for developers who have already updated and are using this id

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**